decisions/0005-product-source-based-required-fields.rst

```diff
@@ -0,0 +1,34 @@
+5. Validating AdditionalMetadata fields based on product source
+--------------------------------------------------------------------
+
+Status
+------
+Accepted (Feb 2023)
+
+Context
+-------
+During the recent influx of external products into the marketplace, a number of products were ingested into publisher which did
+not have all the AdditionalMetadata fields associated to them. These products were coming from different sources and thus had
+different expectations of the validation in AdditionalMetadata fields. As such our current publisher flow, which required all of the
+AdditionalMetadata fields to be present regardless of the product source, needed to be made more flexible to dynamically update its
+validation strategy depending on the particular products.
+
+Decision
+--------
+Modify the AdditionalMetadata field validation logic to base decisions on the source of external products. In particular, have 
+a configuration object detailing the mapping between different product sources and the required additionalMetadata fields for 
```

Comment:
Please add the format we used for configuration and the challenges we faced to map the config to environment var.